dbms/src/Storages/Transaction/RegionBlockReader.cpp

```diff
@@ -37,18 +38,52 @@ RegionBlockReader::RegionBlockReader(DecodingStorageSchemaSnapshotConstPtr schem
     : schema_snapshot{std::move(schema_snapshot_)}
 {}
 
+
 bool RegionBlockReader::read(Block & block, const RegionDataReadInfoList & data_list, bool force_decode)
 {
-    switch (schema_snapshot->pk_type)
+    try
+    {
+        switch (schema_snapshot->pk_type)
+        {
+        case TMTPKType::INT64:
+            return readImpl<TMTPKType::INT64>(block, data_list, force_decode);
+        case TMTPKType::UINT64:
+            return readImpl<TMTPKType::UINT64>(block, data_list, force_decode);
+        case TMTPKType::STRING:
+            return readImpl<TMTPKType::STRING>(block, data_list, force_decode);
+        default:
+            return readImpl<TMTPKType::UNSPECIFIED>(block, data_list, force_decode);
+        }
+    }
+    catch (DB::Exception & exc)
     {
-    case TMTPKType::INT64:
-        return readImpl<TMTPKType::INT64>(block, data_list, force_decode);
-    case TMTPKType::UINT64:
-        return readImpl<TMTPKType::UINT64>(block, data_list, force_decode);
-    case TMTPKType::STRING:
-        return readImpl<TMTPKType::STRING>(block, data_list, force_decode);
-    default:
-        return readImpl<TMTPKType::UNSPECIFIED>(block, data_list, force_decode);
+        // to print more info for debug the random ddl test issue(should serve stably when ddl #004#)
+        // https://github.com/pingcap/tiflash/issues/7024
+        auto print_column_defines = [&](const DM::ColumnDefinesPtr & column_defines) {
+            FmtBuffer fmt_buf;
+            fmt_buf.append(" [column define : ");
+            for (auto const & column_define : *column_defines)
+            {
+                fmt_buf.fmtAppend("(id is {}, name is {}, type is {}) ", column_define.id, column_define.name, column_define.type->getName());
+            }
+            fmt_buf.append(" ];");
+            return fmt_buf.toString();
+        };
+
+        auto print_map = [](auto const & map) {
+            FmtBuffer fmt_buf;
+            fmt_buf.append(" [map info : ");
+            for (auto const & pair : map)
+            {
+                fmt_buf.fmtAppend("(pair.first is {}, pair.second is {}) ", pair.first, pair.second);
+            }
+            fmt_buf.append(" ];");
+            return fmt_buf.toString();
+        };
+
+        exc.addMessage(fmt::format("pk_type is {}, schema_snapshot->sorted_column_id_with_pos is {}, schema_snapshot->column_defines is {}, decoding_snapshot_epoch is {}, block schema is {}", schema_snapshot->pk_type, print_map(schema_snapshot->sorted_column_id_with_pos), print_column_defines(schema_snapshot->column_defines), schema_snapshot->decoding_schema_version, block.dumpJsonStructure()));
```

Comment:
we can also append the `TiKVValue` in `RegionDataReadInfoList` to the message by using `Redact::keyToHexString`